script/deploy

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+# Blue-green deployment script. Usage:
+#
+#   ./script/deploy <appname>
+#
+# Based on
+#   http://www.cloudfoundry.rocks/blue-green-deployment-with-cloudfoundry/
+#   http://docs.pivotal.io/pivotalcf/devguide/deploy-apps/blue-green.html
+#   https://github.com/dlapiduz/step-cloud-foundry-deploy/blob/master/run.sh
+
+set -e
+set -x
+
+BLUE=$1
+GREEN="${BLUE}-B"
+
+# pull the up-to-date manifest from the BLUE (existing) application
+MANIFEST=$(mktemp -t "${BLUE}_manifest")
+cf create-app-manifest $BLUE -p $MANIFEST
+
+# try/catch
+# http://stackoverflow.com/a/22010339/358804
+{
+  DOMAIN=$(cat $MANIFEST | grep domain: | awk '{print $2}')
+
+  # create the GREEN application
+  cf push $GREEN -f $MANIFEST -n $GREEN
+  # ensure it starts
+  curl --fail -I "https://${GREEN}.${DOMAIN}"
```

Comment:
Yeah, apparently `set -e` doesn't apply to functions because _why would it_. 

![rage](http://media.giphy.com/media/mhgKDrTFbVmDK/giphy.gif)

Gave it another pass, using `trap`. Mind trying again?